geom/type_polygon.go

```diff
@@ -253,12 +253,14 @@ func (p Polygon) Envelope() (Envelope, bool) {
 }
 
 func (p Polygon) rings() []LineString {
-	rings := make([]LineString, 1+len(p.holes))
-	rings[0] = p.outer
-	for i, h := range p.holes {
-		rings[1+i] = h
-	}
-	return rings
+	dst := make([]LineString, 0, 1+len(p.holes))
+	return p.appendRings(dst)
```

Comment:
Nit: This call appears like a mutator method on `p`, but is actually a convenience function on `[]LineString`. Feels like both `p` and `dst` should be parameters, and should be swapped to place them in the order they'd appear in the resulting slice i.e. `return appendRings(dst, p)`.

Reply:
Agreed, now that you've pointed it out, it's very obvious (looks like `appendRings` is adding rings to the polygon). Fixed.